src/components/clusterConfiguration/ClusterConfiguration.tsx

```diff
@@ -173,6 +172,10 @@ const ClusterConfiguration: React.FC<ClusterConfigurationProps> = ({ cluster })
 
     // update the cluster validation
     try {
+      if (!values.pullSecret) {
+        // avoid update by backend
+        delete values.pullSecret;
+      }
       const { data } = await patchCluster(cluster.id, values);
       dispatch(updateCluster(data));
```

Comment:
setProvidingPullSecret(false);

Reply:
> I think we should hide the text area every time the form is successfully submitted.

And it already works like that. The user can see pull secret until the form is submitted. Once the cluster is updated (Save draft), its new configuration is returned, newly including `providing_pull_secret: true` which is subsequently passed down to the component which tuns into "hiding" mode - the secret is not shown. 

When clicking on "Change", the secret is not rendered but the user is able to either provide new value or leave it.

Reply:
> Related to comment above, I'd prefer keeping the values below intact and doing this instead:

Yeah, it makes sense to create "a modified copy" right before patching. Will do that.

Reply:
> By doing this we also get ability to reset pull secret field (submit it with '' value) by changing this (line 175) condition to

I think it's is unnecessary logic. Recent use-case can be handled without introducing yet another boolean to drive the flow. In addition, the `pullSecret` is a required param so we should either keep it untouched or pass valid value. If this ever changes, we can tweak the logic round "removing" the field from the cluster-update-params.